### DISCUSSION TITLE
Feature Idea: Ability to omit a column from queries by default

### DISCUSSION THREAD

**Discussion by jhuntdev** [sequelize/sequelize#17069] [Help & Questions]
Mongoose (ORM for MongoDB) has an option on their schemas to prevent a column from being queried unless specifically requested. The most obvious benefit of this feature is that some columns on a table are sensitive (e.g. password), and this protects you if a developer accidentally does a star (*) select on that table and it gets overlooked and deployed.

How this might work in Sequelize:
```js
const User = sequelize.define('User', {
  username: {
    type: DataTypes.STRING,
    primaryKey: true
  },
  password: {
    type: DataTypes.STRING,
    select: false // Mongoose approach - defaults to true (see note below)
  }
});

User.create({
  username: 'sdepold',
  password: 'B7w<3z!@}' // WARNING: For illustration purposes only. Should be securely hashed.
});

const user = await User.findByPk('sdepold');
// { username: 'sdepold' }

const userWithPassword = await User.findByPk('sdepold', {
  attributes: ['username', 'password']
});
// { username: 'sdepold', password: 'B7w<3z!@}' }
```
_Note: If a default-false option is preferred, something like "private: true", "hidden: true", or "exclude: true" could also work._

I've looked for an equivalent option in Sequelize, but haven't found one. I also looked at the source code, but it looks like it might be a little tricky to implement this for someone not well acquainted with the codebase.

**Comment by ephys** [comment]
This would be a nice feature. It would even make it easier to reduce the data being fetched by default

We do have a preference for default-false options (some options are currently default true but we can only do so many breaking changes in one major release). `exclude` is fine, but I can also propose `omitOnSelect`, `noSelect`, or `excludeSelect` as there are other things the attribute could be excluded from (serialization, and validation come to mind)

This option would impact more than `SELECT` queries, the `RETURNING` clause should be impacted by default too
